File: go-concurrency/3-generator/lazy-evaluation.go
package main

import "fmt"

func fibonacciGenerator() <-chan int {
	c := make(chan int)
	go func() {
		for i, j := 0, 1; ; i, j = j, i+j {
			c <- i + j
		}
	}()
	return c
}

func main() {
	c := fibonacciGenerator()

	for i := 0; i < 10; i++ {
		fmt.Println(<-c)
	}
}

File: go-concurrency/3-generator/lazy-evaluation.py
def fibonacciGenerator():
    a, b = 0, 1
    while True:
        a, b = b, a + b
        yield b


if __name__ == '__main__':
    n = 10
    for i in fibonacciGenerator():
        print(i)
        n -= 1
        if n < 0:
            break